TribalWars/src/ApplicationCore/Services/BuildingRequiredService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using ApplicationCore.Entities;
+using ApplicationCore.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ApplicationCore.Services
+{
+    public class BuildingRequiredService : IBuildingRequiredService
```

Comment:
What is the point of this service if it's only calling and returning stuff from repository in untouched form?
It's nice that you are trying to keep the N-tier architecture, but consider making it more meaningful.

Maybe some of the logic (like mulitplying materials and levels) should be part of a business logic (so, a service layer) instead of a repository?
Maybe all facts-checking methods starting with 'Has, Can' could be a part of this layer only and the repository should be focused more on actual database queries?

Reply:
In the repositories I left only the database queries, now the service do the job.